aea/cli/upgrade.py

```diff
@@ -152,6 +168,8 @@ def upgrade_project(ctx: Context) -> None:  # pylint: disable=unused-argument
             upgrader.add_item()
             upgrader.update_references(new_item_version)
 
+        _update_non_vendor_packages(ctx)
```

Comment:
I would have thought this should be called as part of `update_references`. But I guess it works here too..

Reply:
the reason is that `_update_non_vendor_packages` is an expensive operation, since we have to load/dump all the configuration files of the custom packages. 

If I had put the processing in `update_references`, we would have to iterate over all the packages for every item to upgrdae.